tests/test_qbclient_client.py

```diff
@@ -199,3 +199,17 @@ def test_qbclient_test_connection_response_error(qbclient):
         mocked_make_request.return_value = {"msg": "this is not expected"}
         with pytest.raises(ValueError, match="API returned unexpected response"):
             qbclient.test_connection()
+
+
+def test_qbclient_parse_upsert_results_response_success(qbclient, mocked_qb_api_upsert):
+    parsed_response = qbclient.parse_upsert_results(mocked_qb_api_upsert)
```

Comment:
Optional: could skip the variable and directly assert on `qbclient.parse_upsert_results(mocked_qb_api_upsert)`